pkgs/tools/package-management/xbps/default.nix

```diff
@@ -17,14 +17,9 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./cert-paths.patch
-    # fix openssl 3
-    (fetchpatch {
-      url = "https://github.com/void-linux/xbps/commit/db1766986c4389eb7e17c0e0076971b711617ef9.patch";
-      hash = "sha256-CmyZdsHStPsELdEgeJBWIbXIuVeBhv7VYb2uGYxzUWQ=";
-    })
   ];
 
-  env.NIX_CFLAGS_COMPILE = "-Wno-error=unused-result -Wno-error=deprecated-declarations";
+  env.NIX_CFLAGS_COMPILE = "-Wno-error";
```

Comment:
```suggestion
  env.NIX_CFLAGS_COMPILE = "-Wno-error=unused-result -Wno-error=deprecated-declarations";
```

Reply:
```
-Wno-error=unused-result -Wno-cpp
```
also works and does not need the substitute in place.  I opted to leave warnings as warnings to avoid breakage when new compiler versions are used, as the fix for this downstream seems to be to just disable the warnings.

nonetheless, will implement your suggestions. thanks for all the reviews.